uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -9,21 +11,352 @@ define([
+        .then(function (calendar) {
+          vm.calendar = setCalendarProps(calendar);
+        });
+    }
+
+    /**
+     * Loads the leave request day types
+     *
+     * @return {Promise}
+     */
+    function loadDayTypes() {
+      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_day_type')
+        .then(function (dayTypesData) {
+          var iterator,
+            length = dayTypesData.length,
+            typesObj = {};
+
+          // convert to an object with name as key
+          for (iterator = 0; iterator < length; iterator++) {
+            typesObj[dayTypesData[iterator].name] = dayTypesData[iterator];
+          }
+          dayTypes = typesObj;
+        });
+    }
+
+    /**
+     * Loads all the leave requests
+     *
+     * @return {Promise}
+     */
+    function loadLeaveRequest() {
+      return LeaveRequest.all({
+        contact_id: vm.contactId,
+        from_date: {
+          from: vm.selectedPeriod.start_date
+        },
+        to_date: {
+          to: vm.selectedPeriod.end_date
+        }
+      }).then(function (leaveRequestsData) {
+          leaveRequests = leaveRequestsData.list;
+          return loadCalendar();
+        });
+    }
+
+    /**
+     * Loads all the public holidays
+     *
+     * @return {Promise}
+     */
+    function loadPublicHolidays() {
+      return PublicHoliday.all()
+        .then(function (publicHolidaysData) {
+          var iterator,
+            length = publicHolidaysData.length,
+            datesObj = {};
+
+          // convert to an object with time stamp as key
+          for (iterator = 0; iterator < length; iterator++) {
+            datesObj[new Date(publicHolidaysData[iterator].date).getTime()] = publicHolidaysData[iterator];
+          }
+
+          publicHolidays = datesObj;
+        });
+    }
+
+    /**
+     * Loads the status option values
+     *
+     * @return {Promise}
+     */
+    function loadStatuses() {
+      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_status')
+        .then(function (statuses) {
+          var iterator,
+            length = statuses.length,
+            statusesObj = {};
+
+          // convert to an object with value as key
+          for (iterator = 0; iterator < length; iterator++) {
+            statusesObj[statuses[iterator].value] = statuses[iterator];
+          }
+
+          leaveRequestStatuses = statusesObj;
+        });
+    }
 
     /**
-     * Init code
+     * Sets UI related properties(isWeekend, isNonWorkingDay etc)
+     * to the calendar data
+     *
+     * @param  {object} calendar
+     * @return {object}
      */
-    function init() {
-      this.legendCollapsed = false;
+    function setCalendarProps(calendar) {
+      var iterator,
+        dates = Object.keys(calendar.days),
+        length = dates.length,
+        dateObj,
+        leaveRequest;
+
+      for (iterator = 0; iterator < length; iterator++) {
```

Comment:
Same thing as before, i think you can use functional programming style here as well